src/page-modules/assistant/client/journey-planner/index.ts

```diff
@@ -1,16 +1,59 @@
-import { TripData } from '../../server/journey-planner/validators';
-import { TripQuery } from '../../types';
+import useSWR from 'swr';
+import { NonTransitTripData, TripQuery, TripData } from '../../types';
+import { swrFetcher } from '@atb/modules/api-browser';
+import useSWRInfinite from 'swr/infinite';
 
 export type TripApiReturnType = TripData;
+export type NonTransitTripApiReturnType = NonTransitTripData;
 
-export async function nextTripPatterns(
-  query: TripQuery,
-): Promise<TripApiReturnType> {
-  const queryString = tripQueryToQueryString(query);
+function createKeyGetterOfQuery(query: TripQuery) {
+  return (pageIndex: number, previousPageData: TripApiReturnType) => {
+    const cursorKey =
+      query.searchMode === 'arriveBy' ? 'previousPageCursor' : 'nextPageCursor';
+
+    if (previousPageData && !previousPageData[cursorKey]) return null;
+
+    const queryString = tripQueryToQueryString(query);
+
+    // first page, we don't have `previousPageData`
+    if (pageIndex === 0) return `/api/assistant/trip?${queryString}`;
+
+    return `/api/assistant/trip?cursor=${previousPageData[cursorKey]}&${queryString}`;
+  };
+}
+
+export function useTripPatterns(query: TripQuery) {
+  const { data, error, isLoading, isValidating, size, setSize } =
+    useSWRInfinite<TripApiReturnType>(
+      createKeyGetterOfQuery(query),
+      swrFetcher,
+      {},
+    );
 
-  const result = await fetch(`/api/assistant/trip?${queryString}`);
+  const isLoadingInitialData = !data && isLoading;
+  const isLoadingMore = isValidating && size > 1;
+  return {
+    trips: data,
+    isLoading: isLoadingInitialData,
+    isError: error,
```

Comment:
`error` is of type `Error | undefined`, should there be any check so that the boolean value is correct or is it enough that it is defined of `isError` to be correct?

Reply:
Since it is named isError I think this should be a boolean, yes 👍 Updated now. Could return the error as `error` too, but I think this is fine.